server/test/suites/categoryRouter.test.js

```diff
@@ -0,0 +1,169 @@
+      placesId: 'something else',
+      userId: testUser1._id,
+      highlight: 'other highlighted words',
+      comment: 'another sample comment',
+      categoryIds: [testCategory1._id],
+    });
+
+    testCategory1.items = [item1._id, item2._id];
+  });
+
+  after(async () => {
+    await mongoose.disconnect();
+    await mongoServer.stop();
+  });
+
+  /**
+   * Category Router Tests
+   */
+  describe('/createCategory', function () {
+    after(async () => {
+      await Category.remove({});
+    });
+
+    it('add a new category by user2', (done) => {
+      chai
+        .request(server)
+        .post('/category/createCategory')
+        .send({
+          userId: testUser2._id,
+          title: 'User2 New Category',
+        })
+        .set('content-type', 'application/json')
+        .end(async function (error, response) {
+          expect(response).to.have.status(200);
+          try {
+            const categories = await Category.find({});
+            expect(categories).to.have.lengthOf(1);
+            expect(categories[0].title).equals('User2 New Category');
+            done();
+          } catch (error) {
+            expect(error).to.be.null;
+          }
+        });
+    });
+  });
+
+  describe('/getCategories', () => {
+    before(async () => {
+      await testCategory1.save();
+      await testCategory2.save();
+      await testCategory3.save();
+    });
+    after(async () => {
+      await Category.remove({});
+    });
+
+    it('error status and no response when no userId is provided in the request', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategories')
+        .end((error, response) => {
+          expect(response).to.not.have.status(200);
+          expect(response.body).to.be.empty;
+          done();
+        });
+    });
+
+    it('valid request- response body is JSON array of all categories (1) of testUser1', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategories/' + testUser1._id)
+        .end((error, response) => {
+          expect(response).to.have.status(200);
+          expect(response.body).to.have.lengthOf(1);
+          expect(response.body[0].title).equals(testCategory1.title);
+          done();
+        });
+    });
+
+    it('valid request- response body is JSON array of all categories (2) of testUser2', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategories/' + testUser2._id)
+        .end((error, response) => {
+          expect(response).to.have.status(200);
+          expect(response.body).to.have.lengthOf(2);
+          expect(response.body[0].title).equals(testCategory2.title);
+          expect(response.body[1].title).equals(testCategory3.title);
+          done();
+        });
+    });
+
+    it('request userId invalid or doesnt have documents- response body is empty', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategories/' + 'randomUserId')
```

Comment:
Why not just make this one string?

Reply:
Just switched this to an actual wrong id.